src/components/app/artistscontainer/artistslist/artistslistitem/artists-list-item.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react'
+import { observer, inject } from 'mobx-react'
+
+@inject('artistInfoStore')
+@observer
+class ArtistsListItem extends Component {
+  constructor(props) {
+    super(props)
+
+    this.onArtistClick = this.onArtistClick.bind(this)
+  }
+
+  onArtistClick() {
+    const { artistInfoStore, artist } = this.props
+
+    artistInfoStore.updateClickedArtist(artist.id)
+    artistInfoStore.getArtistInfo(artistInfoStore.clickedArtist)
+    artistInfoStore.updateDisplayInfo(true)
+
+    artistInfoStore.getArtistAlbums(artist.id)
+    artistInfoStore.getRelatedArtists(artist.id)
```

Comment:
Nie wiem czy to jest odpowiednie miejsce na wywoływanie tych metod. Nie korzystasz z tego tutaj i ogólnie komponentu ArtistsListItem te zachowanie nie dotyczy i nie powinien on wiedzieć, że takie metody istnieją. Podobny pattern pojawia się też wcześniej, ale napiszę tylko tutaj :P 

Imho lepiej to przenieść do Store lub do komponentu, który tych danych potrzebuje.

Reply:
Do store? W jakim sensie?

Myslalem o tym, zeby bylo w constructorze czy metodzie komponentu, ale bazuja na tym conditional rendery tez. W sensie jak mamy jakies albumy (czyli jak wykona sie getArtistAlbums) to pokazuje jakis komponent dopiero (ktory tego uzywa, wiec do niego mialbym to przeniesc, wiec sie nie wykonywalo ;p).

Reply:
Nom to masz źle skonstruowane :P Komponent ArtistListItem nie powinien wywoływać metod w Storze, których w ogóle nie potrzebuje. Po jakimś czasie wszystko będziesz miał tak poplątane i zależne od siebie, że będzie coraz ciężej dodawać nowe rzeczy lub coś zmieniać.

Wydaje mi się, że dobrym rozwiązaniem będzie zrobić tak, że ArtistListItem będzie wywoływał tylko `updateClickedArtist`. Czy to nie wystarczy do conditional renderu? Jak jest clickedArtist to renderuj jak nie ma to nie renderuj :P I wtedy już komponent bierze to co potrzebuje.

Reply:
Zapisalem sobie do todo bo to bedzie sporo roboty :p

Reply:
Ok ;)